pkg/nuctl/command/get.go

```diff
@@ -99,30 +101,49 @@ func newGetFunctionCommandeer(getCommandeer *getCommandeer) *getFunctionCommande
 				return nil
 			}
 
+			rendererFunc := commandeer.renderFunctionConfig
+			if commandeer.withStatus {
+				rendererFunc = commandeer.renderFunctionConfigWithStatus
+			}
+
 			// render the functions
 			return common.RenderFunctions(commandeer.rootCommandeer.loggerInstance,
 				functions,
 				commandeer.output,
 				cmd.OutOrStdout(),
-				commandeer.renderFunctionConfig)
+				rendererFunc)
 		},
 	}
 
 	cmd.PersistentFlags().StringVarP(&commandeer.getFunctionsOptions.Labels, "labels", "l", "", "Function labels (lbl1=val1[,lbl2=val2,...])")
 	cmd.PersistentFlags().StringVarP(&commandeer.output, "output", "o", common.OutputFormatText, "Output format - \"text\", \"wide\", \"yaml\", or \"json\"")
-
+	cmd.PersistentFlags().BoolVarP(&commandeer.withStatus, "with-status", "s", false, "Whether to return function status along its config")
 	commandeer.cmd = cmd
 
 	return commandeer
 }
 
-func (g *getFunctionCommandeer) renderFunctionConfig(functions []platform.Function, renderer func(interface{}) error) error {
+func (g *getFunctionCommandeer) renderFunctionConfig(functions []platform.Function,
+	renderer func(interface{}) error) error {
 	for _, function := range functions {
 		if err := renderer(function.GetConfig()); err != nil {
 			return errors.Wrap(err, "Failed to render function config")
 		}
 	}
+	return nil
+}
 
+func (g *getFunctionCommandeer) renderFunctionConfigWithStatus(functions []platform.Function,
+	renderer func(interface{}) error) error {
+	for _, function := range functions {
+		functionConfigWithStatus := functionconfig.ConfigWithStatus{
+			Config: *function.GetConfig(),
+			Status: *function.GetStatus(),
+		}
+		if err := renderer(functionConfigWithStatus); err != nil {
+			return errors.Wrap(err, "Failed to render function config with status")
+		}
+	}
```

Comment:
Why not have 1 and always return with status? if some test doesn't care it doesn't have to assert anything on it